lib/spack/llnl/util/tty/log.py

```diff
@@ -321,7 +321,10 @@ def __init__(self, file_like):
     def unwrap(self):
         if self.open:
             if self.file_like:
-                self.file = open(self.file_like, 'w')
+                if sys.version_info < (3,):
```

Comment:
Can we use `self.file = codecs.open(self.file_like, 'w', encoding='utf-8')` for both Py2 and Py3? Maybe even add `errors='ignore'` to drop any invalid characters...